src/strategies/blockzerolabs-cryptonauts/index.ts

```diff
@@ -21,70 +11,13 @@ export async function strategy(
   options,
   snapshot
 ): Promise<Record<string, number>> {
-  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
-
-  // Step 1. Retrieve the total supply of Cryptonauts
-  const totalSupply = options.totalSupply;
-
-  // Step 2. Determine which ones still exist (i.e not burned)
-  const nftExistsMulti = new Multicaller(network, provider, abi, { blockTag });
-  for (let _nftId = 0; _nftId < totalSupply; _nftId++) {
-    nftExistsMulti.call(
-      _nftId,
-      options._nftTokenAddress, // This is a static address
-      'exists',
-      [_nftId]
-    );
-  }
-  const nftExists: Record<number, boolean> = await nftExistsMulti.execute();
-
-  // Step 3. Determine owners for NFTs that still exist
-  const nftOwnersMulti = new Multicaller(network, provider, abi, { blockTag });
-  for (let _nftId = 0; _nftId < totalSupply; _nftId++) {
-    // If the NFT exists, get the owner
-    if (nftExists[_nftId]) {
-      nftOwnersMulti.call(_nftId, options._nftTokenAddress, 'ownerOf', [
-        _nftId
-      ]);
-    }
-  }
-  const nftOwners: Record<number, string> = await nftOwnersMulti.execute();
-
-  // Step 4. Get the balance of each NFT from the Vault
-  const vaultBalanceMulti = new Multicaller(network, provider, abi, {
-    blockTag
-  });
-  for (let _nftId = 0; _nftId < totalSupply; _nftId++) {
-    // If the NFT exists, get the owner
-    if (nftExists[_nftId]) {
-      vaultBalanceMulti.call(_nftId, options._vaultAddress, 'balanceOf', [
-        _nftId
-      ]);
+  const apiUrl = options.api + '?blockNumber=' + snapshot;
+  const response = await fetch(apiUrl, {
+    method: 'GET',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json'
     }
-  }
-  const vaultBalance: Record<
-    number,
-    BigNumberish
-  > = await vaultBalanceMulti.execute();
-
-  // Iterate over each address provided
-  const balances: Record<string, number> = {};
-  addresses.forEach((address) => {
-    let totalBalance = BigNumber.from(0);
-    // Iterate over each NFT
-    for (let _nftId = 0; _nftId < totalSupply; _nftId++) {
-      // Ensure the NFT Exists before continuing to add balance
-      if (nftExists[_nftId]) {
-        // Ensure this address is the owner before continuing to add balance
-        if (nftOwners[_nftId] == getAddress(address)) {
-          // Add the balance
-          totalBalance = totalBalance.add(vaultBalance[_nftId]);
-        }
-      }
-    }
-    // Format the balance with 18 decimals (fixed)
-    balances[address] = parseFloat(formatUnits(totalBalance, options.decimals));
   });
-
-  return balances;
+  return await response.json();
```

Comment:
@umarsa we don't need a separate strategy for this right? we can call the `api` strategy directly right? if API can be modified
https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/strategies/api/index.ts

Reply:
Yeah, we may be able to use that pre-existing strategy.

Do you know what the "addresses" field is sending over to the endpoint?
https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/strategies/api/index.ts#L18

My concern is that the URL might become unnecessarily long and potentially break the RFC guidelines..

Reply:
Hmm yea maybe i think you can try https://snapshot.org/#/strategy/api-post where you can send all addresses in body instead, let me know if it works :)